safesql.go

```diff
@@ -154,6 +180,70 @@ type QueryMethod struct {
 	Param    int
 }
 
+type Issue struct {
+	statement token.Position
+	ignored   bool
+}
+
+// CheckIssues checks lines to see if the line before or the current line has an ignore comment and marks those
+// statements that have the ignore comment on the current line or the line before
+func CheckIssues(lines []token.Position) ([]Issue, error) {
+	files := make(map[string][]token.Position)
+
+	for _, line := range lines {
+		files[line.Filename] = append(files[line.Filename], line)
+	}
+
+	issues := []Issue{}
+
+	for file, linesInFile := range files {
+		// ensure we have the lines in ascending order
+		sort.Slice(linesInFile, func(i, j int) bool { return linesInFile[i].Line < linesInFile[j].Line })
+
+		f, err := os.Open(file)
```

Comment:
can we make this simpler by not using a scanner?  e.g.

```go
contents, err := ioutil.ReadFile(file)
lines := strings.Split(string(contents), "\n")
```
and then just check `HasIgnoreComment(line.Line)` and `BeginsWithComment(line.Line-1) && HasIgnoreComment(line.Line-1)`.

I think this would make the below for loop much much simpler

Reply:
I actually did this to reduce the amount of memory the program could use, ie if a very large file was being processed.

Reply:
sure, but we only process a single file at a time (each time through the loop, the previous file goes out of scope).  Source code files are only 10s of KB in size, and it feels like it would be a big win in terms of readability + simplicity to just read the whole thing in at once.